src/Microsoft.Performance.SDK/Processing/DataSourceGrouping/DataSourceGroup.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Performance.SDK.Processing.DataSourceGrouping
+{
+    /// <inheritdoc/>
+    public sealed class DataSourceGroup
+        : IDataSourceGroup
+    {
+        /// <summary>
+        ///     Initializes a new instance of <see cref="DataSourceGroup"/>.
+        /// </summary>
+        /// <param name="dataSources">
+        ///     The <see cref="IDataSource"/>s in this group. This parameter may not be <c>null</c>.
+        /// </param>
+        /// <param name="processingMode">
+        ///     The <see cref="IProcessingMode"/> that this group should be processed in. This parameter may not
+        ///     be <c>null</c>.
+        /// </param>
+        public DataSourceGroup(IReadOnlyCollection<IDataSource> dataSources, IProcessingMode processingMode)
+        {
+            Guard.NotNull(dataSources, nameof(dataSources));
+            Guard.NotNull(processingMode, nameof(processingMode));
```

Comment:
We haven't used the <exception></exception> doc comments as consistently as we should, but we should try. Since this Guard.NotNull throws, we should add them here. Can copy/paste other examples in code, such as Engine.Create.